orc8r/cloud/go/swagger/combine.go

```diff
@@ -0,0 +1,215 @@
+/*
+ Copyright 2020 The Magma Authors.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree.
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package swagger
+
+import (
+	"regexp"
+	"sort"
+
+	"magma/orc8r/cloud/go/tools/swaggergen/generate"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+
+// CombineStr is a wrapper function around the function Combine()
+// which serves to combine the contents of multiple swagger specs.
+// The output is the merged copy of the swagger spec
+func CombineStr(common string, configs []string) (string, error, error) {
+	cfgs, commonCfg, err := ConvertToSwagger(common, configs)
```

Comment:
We want this entire "combine" process to be resilient against any single service sending garbage. So the CombineToSwagger fn should gracefully handle a service sending non-swagger yaml (or even non-yaml strings). I'll let you figure out a good way to do this